copernicusmarine/download_functions/download_arco_series.py

```diff
@@ -538,7 +538,35 @@ def _download_dataset_as_netcdf(
             contiguous=False,
             shuffle=True,
         )
-        encoding = {var: comp for var in dataset.data_vars}
+        long_encoding = {
+            name: {**var.encoding, **comp}
+            for name, var in dataset.data_vars.items()
+        }
+        encoding = {}
+        for var in dataset.data_vars:
+            keys_to_keep = [
+                "dtype",
+                "shuffle",
+                "blosc_shuffle",
+                "chunksizes",
+                "szip_coding",
+                "zlib",
+                "_FillValue",
+                "significant_digits",
+                "fletcher32",
+                "contiguous",
+                "szip_pixels_per_block",
+                "complevel",
+                "compression_opts",
+                "compression",
+                "quantize_mode",
+                "endian",
+            ]
+            encoding[var] = {
+                key: long_encoding[var][key]
+                for key in long_encoding[var].keys()
+                if key in keys_to_keep
+            }
```

Comment:
```suggestion
        keys_to_keep = {
    "_FillValue",
    "blosc_shuffle",
    "chunksizes",
    "complevel",
    "compression",
    "compression_opts",
    "contiguous",
    "dtype",
    "endian",
    "fletcher32",
    "quantize_mode",
    "shuffle",
    "significant_digits",
    "szip_coding",
    "szip_pixels_per_block",
    "zlib",
}
        encoding = {
            name: {**{key: value for key, value in var.encoding if key in keys_to_keep}, **comp}
            for name, var in dataset.data_vars.items()
        }
```

To be formatted and check if it works but the idea is that this part of the code can be more concise. (also notice the user of a set and not a list for the `keys_to_keep`

Reply:
I feel like this is complicating too much the code (and to be honest it will be hard for me to actually make it work). Do you think it is necessary to have the code as concise as possible? (even if it get's more tricky to maintain and develop 🌚 ?)

Reply:
Mmh okay 🫡